hack/release.sh

```diff
@@ -114,6 +114,92 @@ s3_url() {
 	esac
 }
 
+map_goos_to_uname() {
+	# we need to map our GOOS to uname values
+	# see https://en.wikipedia.org/wiki/Uname
+	# ie, GOOS=linux -> "uname -s"=Linux
+	GOOS=$1
+	case "$GOOS" in
+		darwin)
+			echo "Darwin"
+			;;
+		freebsd)
+			echo "FreeBSD"
+			;;
+		linux)
+			echo "Linux"
+			;;
+		windows)
+			# this is windows use the .zip and .exe extensions for the files.
+			echo "Windows"
+			;;
+		*)
+			echo >&2 "error: can't convert $GOOS to an appropriate value for 'uname -s'"
+			exit 1
+			;;
+	esac
+}
+
+map_goarch_to_uname() {
+	# we need to map our GOARCH to uname values
+	# see https://en.wikipedia.org/wiki/Uname
+	# ie, GOARCH=amd64 -> "uname -m"=x86_64
+	GOARCH=$1
+	case "$GOARCH" in
+		amd64)
+			echo "x86_64"
+			;;
+		386)
+			echo "i386"
+			;;
+		arm)
+			echo "armel"
+			# someday, we might potentially support multiple GOARM values, in which case we might get armhf here too
+			;;
+		*)
+			echo >&2 "error: can't convert $GOARCH to an appropriate value for 'uname -m'"
+			exit 1
+			;;
+	esac
+}
+
+get_release_entry() {
+	GOOS=$1
+	GOARCH=$2
+	s3Os=`map_goos_to_uname $GOOS`
+	s3Arch=`map_goarch_to_uname $GOARCH`
+	zipExt=".tgz"
```

Comment:
All these variables ought to be `local`, and `$()` should be preferred over backticks, ie:
```bash
get_release_entry() {
	local GOOS="$1"
	local GOARCH="$2"
	local s3Os="$(map_goos_to_uname "$GOOS")"
	local s3Arch="$(map_goarch_to_uname "$GOARCH")"
	local zipExt='.tgz'
	...
```